src/ir_York.cpp

```diff
@@ -0,0 +1,358 @@
+using irutils::minsToString;
+
+
+// Constants
+const uint16_t kHdrMark = 4887;
+const uint16_t kBitMark = 612;
+const uint16_t kHdrSpace = 2267;
+const uint16_t kOneSpace = 1778;
+const uint16_t kZeroSpace = 579;
+const uint16_t kFreq = 38000;  // Hz. (Guessing the most common frequency.)
+const uint16_t kOverhead = 3;
+
+#if SEND_YORK
+// Alternative >64bit function to send YORK messages
+// Function should be safe over 64 bits.
+/// Send a  formatted message.
+/// Status: ALPHA / Untested. The Checksum logic is missing
+/// @param[in] data An array of bytes containing the IR command.
+///                 It is assumed to be in MSB order for this code.
+/// e.g.
+/// @code
+///   uint8_t data[kStateLength] = {0x10, 0x08, 0xE0, 0x40, 0x02, 0x10, 0x80, 0x10, 0x80, 0x12, 0x00, 0x00, 0x00, 0x00, 0x37, 0xA0, 0x37};
+/// @endcode
+/// @param[in] nbytes Nr. of bytes of data in the array. (>=kStateLength)
+/// @param[in] repeat Nr. of times the message is to be repeated.
+void IRsend::sendYork(const uint8_t data[], const uint16_t nbytes, const uint16_t repeat) {
+  for (uint16_t r = 0; r <= repeat; r++) {
+    uint16_t pos = 0;
+    sendGeneric(kHdrMark, kHdrSpace,
+                kBitMark, kOneSpace,
+                kBitMark, kZeroSpace,
+                kBitMark, kDefaultMessageGap,
+                data + pos, 17,  // Bytes
+                kFreq, false, kNoRepeat, kDutyDefault); //false == LSB
+    pos += 17;  // Adjust by how many bytes of data we sent
+  }
+}
+#endif  // SEND_YORK
+
+
+#if DECODE_YORK
+/// Decode the supplied  message.
+/// Status: ALPHA / Tested, some values still are not mapped to the internal state of AC
+/// @param[in,out] results Ptr to the data to decode & where to store the decode
+/// @param[in] offset The starting index to use when attempting to decode the
+///   raw data. Typically/Defaults to kStartOffset.
+/// @param[in] nbits The number of data bits to expect.
+/// @param[in] strict Flag indicating if we should perform strict matching.
+/// @return A boolean. True if it can decode it, false if it can't.
+bool IRrecv::decodeYork(decode_results *results, uint16_t offset, const uint16_t nbits, const bool strict) {
+  if (results->rawlen < 2 * nbits + kOverhead - offset)
+    return false;  // Too short a message to match.
+  if (strict && nbits != kYorkBits)
+    return false;
+
+  uint16_t pos = 0;
+  uint16_t used = 0;
+
+  used = matchGeneric(results->rawbuf + offset, results->state + pos,
+                      results->rawlen - offset, 136,
+                      kHdrMark, kHdrSpace,
+                      kBitMark, kOneSpace,
+                      kBitMark, kZeroSpace,
+                      kBitMark, kDefaultMessageGap,
+                      false, _tolerance, kMarkExcess,
+                      false); //LSB
+  if (used == 0) return false;  // We failed to find any data.
+  offset += used;  // Adjust for how much of the message we read.
+  pos += 17;  // Adjust by how many bytes of data we read
```

Comment:
You can remove these two lines. They were automatically generated code that isn't required in this situation.